doc/whats_new/v1.3.rst

```diff
@@ -155,6 +155,9 @@ Changelog
 - |Fix| :func:`metric.manhattan_distances` now supports readonly sparse datasets.
   :pr:`25432` by :user:`Julien Jerphanion <jjerphan>`.
 
+- |Enhancement| :func:`metric.haversine_distances` can now compute pairwise distances.
+  :pr:`22705` by :user:`Jona Sassenhagen <jona-sassenhagen>` and :user:`Charles Patel <acharles7>`.
+
```

Comment:
This entry is unrelated to the scope of this PR. It must be changed to use an `|API|` one stating the introduction of `paired_haversine_distances`.